### DISCUSSION TITLE
SpringCloudGateway3.1.3, With apm-log4j2-2.x:8.10.0, does not display the real traceId, always display 'TID: N/A'

### DISCUSSION THREAD

**Discussion by ye213** [apache/skywalking#9232] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

Use the log4j2, with agent version 8.10.0, in the log, the tid always display as 'TID: N/A'
like `[2022-06-15 14:53:19.958] [TID: N/A]`
but the other service through the gateway has the correct traceId。And we can see the trace in the skywalkig ui
![image](https://user-images.githubusercontent.com/19651059/173783671-ce3be7cc-7c99-424c-96d4-ef115057071e.png)

### What you expected to happen

the SpringCloudGateway which use log4j2 print the right traceId

### How to reproduce

SpringCloudGateway3.1.3
apm-log4j2-2.x:8.10.0
Skywalking 9.0.0
log4j2 xml pattern: `[%d{yyyy-MM-dd HH:mm:ss.SSS}] [%traceId] [%logger{36}] [%thread] [%-5level] %msg%n`

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
The plugin doesn't promise you to have trace ID in the logs, it just does tracing. So this is not a bug as it is not a feature.

Trace ID is available when and only when the tracing context exists in the current thread. Spring Gateway provides a complex async module, so, your codes may run in another thread. This is a very normal case.

**Reply by ye213** [reply]
I think the traceId can get from the **ContextManager.capture()**,  just like **DispatcherHandlerHandleMethodInterceptor** in the  webflux-5.x plugin
there is another question, why delete it ?

**Reply by wu-sheng** [reply]
> I think the traceId can get from the ContextManager.capture(), just like DispatcherHandlerHandleMethodInterceptor in the webflux-5.x plugin

I didn't say it can't. 
The community works if someone contributed this feature. You are using this project contributed by 600+ global contributors. Those features were there from the beginning.

**Comment by yugj** [comment]
inelegant way:

    <dependency>
      <groupId>org.apache.skywalking</groupId>
      <artifactId>apm-toolkit-trace</artifactId>
    </dependency>

using RunnableWrapper, ConsumerWrapper to wrap spring cloud gateway async code, to let skywalking know the context
